queries/cow_rewards.sql

```diff
@@ -10,59 +10,17 @@ with trade_hashes as (SELECT solver,
 
-     final_counts as (select concat('0x', encode(solver, 'hex')) as receiver,
-                             count(tx_hash)                      as num_batches,
-                             sum(num_trades)                     as num_trades,
-                             -- There represent JIT liquidity orders
-                             sum(missing_orders)                 as jit_orders
-                      from corrected_trade_counts
-                      group by solver)
-
-select *,
-       ((50 * num_batches + 35 * (num_trades - jit_orders)) * pow(10, 18))::numeric::text as amount,
-       '0xDEf1CA1fb7FBcDC777520aa7f396b4E015F497aB'                        as token_address
-from final_counts
-order by receiver;
-
--- There is currently a bug in orderbook (missing entries in solver_competitions table)
--- with solver_solutions as (select concat('0x', encode(solver, 'hex'))    as receiver,
---
---                                  count(*)                               as num_batches,
---                                  -- winning solution is the last element of the array solutions array.
---                                  -- 1. extract solutions from json,
---                                  -- 2. pop the last element of the array
---                                  -- 3. and extract the length of the orders in it
---                                  sum(jsonb_array_length(json -> 'solutions' ->
---                                                         jsonb_array_length(json -> 'solutions') -
---                                                         1 -> 'orders')) as num_trades
---                           from solver_competitions sc
---                                    inner join settlements s
---                                               on s.tx_hash = sc.tx_hash
---                           where s.block_number between {{start_block}} and {{end_block}}
---                           group by solver)
---
--- select '0xDEf1CA1fb7FBcDC777520aa7f396b4E015F497aB'                        as token_address,
---        receiver,
---        ((50 * num_batches + 35 * num_trades) * pow(10, 18))::numeric::text as amount
--- from solver_solutions
--- order by receiver;
\ No newline at end of file
+select concat('0x', encode(solver, 'hex'))          as receiver,
+       -- This column will be used to compare with Dune:
+       -- cross referencing that all trades are accounted for in the orderbook
+       count(*) as num_trades,
+       (sum(reward) * pow(10, 18))::numeric::text   as amount,
+       '0xDEf1CA1fb7FBcDC777520aa7f396b4E015F497aB' as token_address
+from trade_hashes
+         -- Inner join because both prod and staging DB index trades and settlements,
+-- but the rewards for each environment are contained only in respective databases.
+         inner join order_rewards
+                    on trade_hashes.order_uid = order_rewards.order_uid
```

Comment:
Don't we also need to join on the auction_id? The order_uid is not unique, if we don't have constant orders rewards, right?

Reply:
Good thinking. I'll try it out and see.

Reply:
Is this what you were thinking?

https://github.com/cowprotocol/solver-rewards/pull/96/commits/981e6bdbca963921071300c603b35718cedb0189